src/bosh-director/spec/unit/problem_resolver_spec.rb

```diff
@@ -9,12 +9,31 @@ module Bosh::Director
     let(:task) { Bosh::Director::Models::Task.make(id: 42, username: 'user') }
     let(:task_writer) { Bosh::Director::TaskDBWriter.new(:event_output, task.id) }
     let(:event_log) { Bosh::Director::EventLog::Log.new(task_writer) }
+    let(:update_deployment) { double }
+    let(:parallel_problem_resolution) { true }
+    let(:parallel_update_config) { instance_double('Bosh::Director::DeploymentPlan::UpdateConfig', serial?: false) }
+    let(:n_igs_with_problems) { 4 }
+    let(:n_problems_in_ig) { 3 }
```

Comment:
It's not obvious what `n_problems_in_ig` means, and it's easily confused with the above variable `n_igs_with_problems`. Would you consider renaming to something like `problems_per_instance_group` and `num_problem_instance_groups `?